packages/pluggableWidgets/carousel-web/src/Carousel.tsx

```diff
@@ -0,0 +1,39 @@
+import { createElement, ReactNode, useEffect, useState, useCallback, ReactElement } from "react";
+import { executeAction } from "@mendix/piw-utils-internal";
+import { CarouselContainerProps } from "../typings/CarouselProps";
+import { Carousel as CarouselComponent } from "./components/Carousel";
+import loadingCircleSvg from "./ui/loading-circle.svg";
+import "./ui/Carousel.scss";
+
+export function Carousel(props: CarouselContainerProps): ReactNode {
+    const { dataSource, content, showPagination, loop, tabIndex, navigation, animation, delay, autoplay } = props;
+    const [loading, setLoading] = useState(true);
```

Comment:
Is there a particular reason you're creating a new state variable to track loading instead of using `dataSource?.status`? It already contains a loading state and it feels like this variable is basically the same as that one.

Reply:
Just a readable way to differentiate between "not available" and "available" state

Reply:
The problem is that it introduces additional and unnecessary re-renders for the same thing. If `dataSource?.status` now changes from loading to available, that in itself already causes a re-render. But because of this state variable, you also do a `setLoading(false)`, which also causes a re-render, even though they're the same thing.  This is actually bad for performance.

You're only using this `loading` variable in the ternary on L45. You can either do `const loading = dataSource?.status !== ValueStatus.Available` or just inline the conditional there.